gpii/node_modules/flowManager/src/SolutionsRegistryDataSource.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * GPII Solutions Registry Datasource
+ *
+ * Copyright 2016 RtF-I
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013)
+ * under grant agreement no. 289016.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    fs = require("fs");
+
+
+require("kettle");
+
+fluid.registerNamespace("gpii.flowManager.solutionsRegistry");
+
+fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource", {
+    gradeNames: ["kettle.dataSource"],
+    members: {
+        fullSolutionsRegistry: null
+    },
+    readOnlyGrade: "gpii.flowManager.solutionsRegistry.dataSource",
+    invokers: {
+        getImpl: {
+            funcName: "gpii.flowManager.solutionsRegistry.dataSource.handle",
+            args: ["{that}", "{arguments}.1", "{arguments}.2"] // options, directModel
+        }
+    },
+    listeners: {
+        onCreate: "gpii.flowManager.solutionsRegistry.dataSource.loadSolutionsRegistry"
+    }
+});
+
+gpii.flowManager.solutionsRegistry.dataSource.loadSolutionsRegistry = function (that) {
+    if (!that.options.path) {
+        fluid.fail("The solutionsRegistry datasource ", that, " needs a \"path\" option pointing to the solution entries folder");
+    }
+    var url = fluid.module.resolvePath(that.options.path);
+    if (!fs.existsSync(url)) {
+        fluid.fail("The path provided to the solutionsRegistry datasource (", url, ") has not been found on the file system");
+    }
+    that.fullSolutionsRegistry = require(url);
+};
+
+fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource.writable", {
+    gradeNames: ["kettle.dataSource.writable"],
+    invokers: {
+        setImpl: {
```

Comment:
Bit dodgy advertising a "set" method when it doesn't work! Best to remove this grade